### DISCUSSION TITLE
Autoscaling compatible?

### DISCUSSION THREAD

**Discussion by codeagencybe** [kube-hetzner/terraform-hcloud-kube-hetzner#648] [Q&A]
Hello

To what version or from which version is autoscaling nodes/cluster supported? 
I found the docs to be very confusing about this. 

AS feature is supported for a long while now, but from what I read on rancher docs, they claim AS feature only works from Kubernetes 1.25??
While the current stable Kubernetes version is still 1.24.x AFAIK. 
And Rancher docs says it's only supported up to 1.24; not 1.25. 
Then how they can claim AS support with this weird combination? 

My goals:
Rancher 2.7.1 (current stable version)
K3s
Autoscaling nodes enabled

Which kubernetes version? 1.24 or 1.25? And does auto scaling nodes work under 1.24? Or does it really need 1.25? 
Anybody who has running this combination? Does it work fine? Any problems? 

@mysticaltech do you have any idea about this? 

Thanks!

**Comment by mysticaltech** [comment]
@codeagencybe I missed your tag somehow. Yes, we use v1.25 by default and it works great with the autoscaler.

**Comment by codeagencybe** [comment]
@mysticaltech 
Also with Rancher installed? Because that's not working. If you try Kubernetes 1.25 or higher, then Rancher complaints about compatibility. I did a test also with Rancher desktop on a local setup, same problem. 
And I also got confirmation from Rancher itself too about this. 

<img width="397" alt="image" src="https://user-images.githubusercontent.com/7568546/225853189-7bc7c3c5-1392-481d-a004-d7e970a2ed2b.png">

**Comment by codeagencybe** [comment]
From local setup with Rancher desktop I also get this error: 

```

helm install rancher rancher-latest/rancher --namespace cattle-system --set hostname=rancher.local --set bootstrapPassword=xxx --wait --timeout=10m
Error: INSTALLATION FAILED: chart requires kubeVersion: < 1.25.0-0 which is incompatible with Kubernetes v1.25.7+k3s1
```

**Reply by mysticaltech** [reply]
Ah I see, so move everything to 1.24, grep 1.25 in the variables.tf to see which variables to change via your kube.tf. It includes the autoscaler.
